docs/Displaying a User Interface/Alerts/index.md

```diff
@@ -8,128 +8,205 @@ The alert will persist on the screen until the timeout has elapsed, or the user
 
-// Text-to-speech
-alert.ttsChunks = SDLTTSChunk.textChunks(from: "<#Text to speak#>")
+@![iOS]
+#### Objective-C
+```objc
+alert.progressIndicator = @YES;
+```
 
-// Special tone played before the tts is spoken
-alert.playTone = true
+#### Swift
+```swift
+alert.progressIndicator = true
+```
+!@
 
-// Soft buttons
-let okButton = SDLSoftButton()
-okButton.text = "OK"
-okButton.type = .text
-okButton.softButtonID = <#Soft Button Id#>
-okButton.handler = { (buttonPress, buttonEvent) in
-    guard let press = buttonPress else { return }
+@![android,javaSE,javaEE]
+` TODO - code example `
+!@
 
-    // create a custom action for the selected button
-}
+### Text-To-Speach
+The alert can also be formatted to speak a prompt when the alert appears on the screen. Do this by setting the `ttsChunks` parameter.
 
-alert.softButtons = [okButton]
+@![iOS]
 
-// Send the alert
-sdlManager.send(request: alert) { (request, response, error) in
-    if response?.resultCode == .success {
-        // alert was dismissed successfully
-    }
-}
+#### Objective-C
+```objc
+alert.ttsChunks = [SDLTTSChunk textChunksFromString:@"<#Text to speak#>"];
+```
+
+#### Swift
+```swift
+alert.ttsChunks = SDLTTSChunk.textChunks(from: "<#Text to speak#>")
 ```
 !@
 
-@![android, javaSE, javaEE]
-```java
-Alert alert = new Alert();
-alert.setAlertText1("Alert Text 1");
-alert.setAlertText2("Alert Text 2");
-alert.setAlertText3("Alert Text 3");
+@![android,javaSE,javaEE]
+` TODO - code example `
+!@
 
-// Maximum time alert appears before being dismissed
-// Timeouts must be between 3-10 seconds
-// Timeouts may not work when soft buttons are also used in the alert
-alert.setDuration(5000);
+### Play Tone
+To play the alert tone before the text-to-speech is spoken, set `playTone` to `true`.
 
-// A progress indicator (e.g. spinning wheel or hourglass)
-// Not all head units support the progress indicator
-alert.setProgressIndicator(true);
+@![iOS]
+#### Objective-C
+```objc
+alert.playTone = @YES;
+```
 
-// Text-to-speech
-alert.setTtsChunks(TTS_list); // TTS_list populated elsewhere
+#### Swift
+```swift
+alert.playTone = true
+```
+!@
 
-// Special tone played before the tts is spoken
-alert.setPlayTone(true);
+@![android,javaSE,javaEE]
+` TODO - code example `
+!@
 
-// Soft buttons
-alert.setSoftButtons(softButtons); // softButtons populated elsewhere
+### Show The Alert
+@![iOS]
+#### Objective-C
+```objc
+[self.sdlManager sendRequest:alert withResponseHandler:^(SDLRPCRequest *request, SDLRPCResponse *response, NSError *error) {
+    if ([response.resultCode isEqualToEnum:SDLResultSuccess]) {
+        // alert was dismissed successfully
+        }
+}];
+```
 
-// Send alert
-sdlManager.sendRPC(alert);
+#### Swift
+```swift
+sdlManager.send(request: alert) { (request, response, error) in
+    if response?.resultCode == .success {
```

Comment:
Look at other guides for how this is properly done with a guard. Maintain consistency across documentation.

Reply:
fixed